.github/copilot-instructions.md

```diff
@@ -0,0 +1,150 @@
+
+### High Level Repository Information
+- **Type**: Kotlin Multiplatform library project
+- **Size**: Medium-sized (23 modules across core and extension libraries)
+- **Languages**: Kotlin (primary), Gradle Kotlin DSL for build scripts
+- **Target Platforms**: JVM, JS (browser), iOS (`iosX64`, `iosArm64`, `iosSimulatorArm64`), WebAssembly JS
+- **Build System**: Gradle with custom build plugins
+- **Documentation**: Generated with Dokka
+- **Testing**: Kotlin Test with Kotest property testing
+
+## Build Instructions
+
+### Prerequisites
+- **JDK 8 (Zulu distribution)** - Required for building and testing (as specified in CI workflows)
+- **macOS environment** recommended for full multiplatform support (Linux has limited iOS target support)
+- Initial setup time: ~2-3 minutes for Gradle daemon initialization and dependency download
+
+### Environment Setup
+Always ensure JDK 8 is properly configured before building. The project uses Gradle wrapper, so no manual Gradle installation is needed.
+
+**IMPORTANT**: On non-macOS systems, iOS targets (`iosArm64`, `iosSimulatorArm64`, `iosX64`) will be disabled. Add `kotlin.native.ignoreDisabledTargets=true` to suppress warnings.
+
+### Bootstrap and Build Commands
+
+#### Essential Commands (validated and working):
+1. **Clean the project**: `./gradlew clean` (takes ~2-3 seconds after initial setup)
+2. **Build all modules**: `./gradlew build` (takes 5-10 minutes initially)
+3. **Run verification checks**: `./gradlew check` (comprehensive testing, takes 5-15 minutes)
+4. **Publish to local Maven**: `./gradlew publishToMavenLocal` (recommended for testing changes)
+
+#### Testing Commands:
+- **Run JVM tests for a module**: `./gradlew :kotlin-common-[module]:jvmTest`
+- **Run JS tests for a module**: `./gradlew :kotlin-common-[module]:jsTest` 
+- **Run all tests for a module**: `./gradlew :kotlin-common-[module]:allTests`
+- **Global test command**: There is no single `test` task - use `check` for comprehensive verification
+
+#### Documentation:
+- **Generate API docs**: `./gradlew :dokkaGeneratePublicationHtml`
+- **Output location**: `build/dokka/html/`
+
+### Build Timing and Known Issues
+
+**Timing Expectations**:
+- First build: 5-10 minutes (includes dependency resolution)
+- Subsequent builds: 30 seconds - 2 minutes
+- Test execution: 2-10 minutes depending on scope
+- Clean builds: Add 1-2 minutes
+
+**Common Warnings (can be ignored)**:
+- iOS target warnings on non-macOS systems
+- WebAssembly environment selection warnings
+- Dokka experimental plugin warnings
+- Gradle deprecation warnings (project targets Gradle 8.11.1)
+
+**Error Handling**:
+- If build fails with "Cannot locate tasks that match 'test'", use target-specific tasks like `jvmTest`, `jsTest`
```

Comment:
See the comment on tests above. You should not use the `test` task in multiplatform modules.